lib/runtime/console.coffee

```diff
@@ -72,11 +72,15 @@ module.exports =
   info: (data) -> @c.info data
 
   toggle: ->
-    d = atom.workspace.paneContainerForItem(@c)
-    if !d
+    dock = atom.workspace.paneContainerForItem(@c)
+    if dock and dock.toggle
+      if dock.isVisible()
+        dock.hide()
+      else
+        dock.show()
+        @c.open()
```

Comment:
There's a [new API](http://blog.atom.io/2017/05/23/docks-deep-dive.html) for this, so it should be possible to do `atom.workspace.toggle(@c)`. I guess you'll want to avoid that for compatibility but might be worth making a note of